### DISCUSSION TITLE
Async onError in in QueryCache/MutationCache

### DISCUSSION THREAD

**Discussion by szajcz** [TanStack/query#8166] [Q&A]
Hi everybody,

I'm setting up a global QueryCache and MutationCache in my application to achieve one thing: signing out the user when a specific error status is received. This includes calling an async function (namely `deleteItemAsync` from expo-secure-store).

As the `onError` callback on both QueryCache and MutationCache is a synchronous function, I'm questioning if I'm taking the right approach here. Is there a better way/place to make the async call? I assume it's a pretty common case during error handling.

Thanks in advance (and thank you for this amazing library)!

**Comment by TkDodo** [comment]
I think you can just call your async function without awaiting it
